### PR TITLE
Fixes #508 wand of the war mage missing bonuses

### LINKED ISSUE #508: Wand of the War Mage does not apply its bonuses
**Describe the bug**
Wand of the War Mage should work similar to Rod of the Pact Keeper, but does not. No bonuses are applied.

**To Reproduce**

1. Equip Wand of the War Mage to an appropriate spell caster
2. Notice no change to stats

**Expected behavior**

Spell Attacks should be boosted by (+1/+2/+3)

### PR DESCRIPTION
**Pull request by codeGlaze**
## Description:

**Related issue (if applicable):** fixes #508
Wand of the War Mage was not applying the spell attack modifier.

**WARNING:  breaks existing name keys for the various Wands of the War Mage**

+ Introduce new generic function for creating items (focused on caster bonuses but could be generic)

+ Added some missing "action" blurbs

+ Fixes possible typo in `from-internal-item` properties

+ Automatic EOL changes to fit file format

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] I have commented my code, particularly in hard-to-understand areas
  - There **is** commented out code in this PR. Features to return to or tweak, primarily documentation related.
  - [x] My changes generate no new warnings (check the console)

### CODE REVIEW

File: src/cljc/orcpub/dnd/e5/magic_items.cljc [hunk 1]
```diff
@@ -333,6 +334,25 @@ Additionally, you can focus your senses as an action to magically discern the di
 direction to the closest dragon within 30 miles of you that is of the same type as the armor. This special action can’t be used again until the next dawn."
                      )})
 
+(defn
```
Comment (datdamnzotz) [inline src/cljc/orcpub/dnd/e5/magic_items.cljc:337]: There already is a `spell-attack-modifier-bonus` and a `spell-save-dc-bonus` in modifiers.cljc
Do they not provide the functionality your after?

They can be used like this:
```
::modifiers [(mod5e/spell-attack-modifier-bonus 2)
(mod5e/spell-save-dc-bonus 3)]
```

File: src/cljc/orcpub/dnd/e5/magic_items.cljc [hunk 2]
```diff
@@ -2816,25 +2844,17 @@ The wand regains 1d6 + 1 expended charges daily at dawn. If you expend the wand
      ::type :wand
      ::rarity :uncommon
      ::description "The wand has 3 charges. While holding it, you can use an action to expend 1 of its charges, and if a secret door or trap is within 30 feet of you, the wand pulses and points at the one nearest to you. The wand regains 1d3 expended charges daily at dawn."
-     }{
```
Comment (datdamnzotz) [inline src/cljc/orcpub/dnd/e5/magic_items.cljc:2819]: I think you just need to add:

File: src/cljc/orcpub/dnd/e5/magic_items.cljc [hunk 3]
```diff
@@ -2816,25 +2844,17 @@ The wand regains 1d6 + 1 expended charges daily at dawn. If you expend the wand
...
-     name-key "Wand of the War Mage, +1"
-     ::type :wand
-     ::rarity :uncommon
-     ::attunement [:spellcaster]
```
Comment (datdamnzotz) [inline src/cljc/orcpub/dnd/e5/magic_items.cljc:2823]:
```
::modifiers [(mod5e/spell-attack-modifier-bonus 1)
(mod5e/spell-save-dc-bonus 1)]
```

File: src/cljc/orcpub/dnd/e5/magic_items.cljc [hunk 4]
```diff
@@ -2816,25 +2844,17 @@ The wand regains 1d6 + 1 expended charges daily at dawn. If you expend the wand
...
-     ::description "While holding this wand, you gain a +1 bonus to spell attack rolls. In addition, you ignore half cover when making a spell attack."
-     }{
-     name-key "Wand of the War Mage, +2"
-     ::type :wand
-     ::rarity :rare
-     ::attunement [:spellcaster]
```
Comment (datdamnzotz) [inline src/cljc/orcpub/dnd/e5/magic_items.cljc:2829]:
```
::modifiers [(mod5e/spell-attack-modifier-bonus 2)
(mod5e/spell-save-dc-bonus 2)]
```

File: src/cljc/orcpub/dnd/e5/magic_items.cljc [hunk 5]
```diff
@@ -2816,25 +2844,17 @@ The wand regains 1d6 + 1 expended charges daily at dawn. If you expend the wand
...
-     ::description "While holding this wand, you gain a +2 bonus to spell attack rolls. In addition, you ignore half cover when making a spell attack."
-     }{
-     name-key "Wand of the War Mage, +3"
-     ::type :wand
-     ::rarity :very-rare
-     ::attunement [:spellcaster]
```
Comment (datdamnzotz) [inline src/cljc/orcpub/dnd/e5/magic_items.cljc:2835]:
```
::modifiers [(mod5e/spell-attack-modifier-bonus 3)
(mod5e/spell-save-dc-bonus 3)]
```

File: src/cljc/orcpub/dnd/e5/magic_items.cljc [hunk 6]
```diff
@@ -2816,25 +2844,17 @@ The wand regains 1d6 + 1 expended charges daily at dawn. If you expend the wand
...
-     ::description "While holding this wand, you gain a +3 bonus to spell attack rolls. In addition, you ignore half cover when making a spell attack."
-     }{
+     }
```
Comment (datdamnzotz) [inline src/cljc/orcpub/dnd/e5/magic_items.cljc:2847]: and this whole section can be removed.

**Review by datdamnzotz — CHANGES_REQUESTED** [review]
See inline comments

**Comment by codeGlaze** [comment]
That's exactly what I did.

I used those modifiers, just abstracted it to cut down boilerplate code and potentially reuse for other items/custom items in the future.

**Comment by datdamnzotz** [comment]
> That's exactly what I did.
> 
> I used those modifiers, just abstracted it to cut down boilerplate code and potentially reuse for other items/custom items in the future.

Looks like this is moving away from the convention that is already in place across all the magic items for three items.
This is really the data set for all magic items, they should follow the same syntax across all of them docha think?

I disagree with what he did with the `horn-of-valhalla` or `figurine-of-wondrous-power` if you search those items they are defined by a function.  There are a few of those.
They really should just be typed out like the rest instead of pushing functions for them.

I'm not apposed to introducing something like this - but imho it is the data set, probably should just follow that architecture - now if you want to update them all that's fine, but let's not just do 3.

**Comment by datdamnzotz** [comment]
In https://dev.dungeonmastersvault.com for testing

**Pull request merged**